components/org.wso2.carbon.identity.conditional.auth.functions.cookie/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/cookie/CookieFunctionImpl.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.carbon.identity.conditional.auth.functions.cookie;
+
+import org.apache.axiom.om.util.Base64;
+import org.apache.commons.io.Charsets;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.carbon.core.util.CryptoException;
+import org.wso2.carbon.core.util.CryptoUtil;
+import org.wso2.carbon.core.util.SignatureUtil;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsServletRequest;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsServletResponse;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.conditional.auth.functions.cookie.util.Constants;
+
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.http.Cookie;
+
+/**
+ * Implementation of the setCookie and getCookieValue functions.
+ */
+public class CookieFunctionImpl implements SetCookieFunction, GetCookieFunction {
+
+    private static final Log log = LogFactory.getLog(CookieFunctionImpl.class);
+
+    @Override
+    public void setCookie(JsServletResponse response, String name, String value, Map<String, Object> properties) {
+
+        boolean sign = Optional.ofNullable((Boolean) properties.get(Constants.SIGN)).orElse(false);
+        boolean encrypt = Optional.ofNullable((Boolean) properties.get(Constants.ENCRYPT)).orElse(false);
+        String signature = null;
+        if (sign) {
+            try {
+                signature = Base64.encode(SignatureUtil.doSignature(value));
+            } catch (Exception e) {
+                log.error("Error occurred when signing the cookie value.", e);
+                return;
+            }
+        }
+        if (encrypt) {
+            try {
+                value = CryptoUtil.getDefaultCryptoUtil().encryptAndBase64Encode(Base64.decode(value));
```

Comment:
Also, shall we check for the value length before trying to encrypt

Reply:
maximum allowed value length will verify with the key length. If the value length is larger than the key length, the encryptAndBase64Encode() will throw an exception. We caught it and log the error.